include/aspect/simulator.h

```diff
@@ -1060,6 +1072,14 @@ namespace aspect
        * @{
        */
       Parameters<dim>                     parameters;
+
+      /**
+       * Shared pointer for an instance of the MeltHandler. This way,
+       * if we do not need the machinery for doing melt stuff, we do
+       * not even allocate it.
+       */
+      std_cxx11::shared_ptr<MeltHandler<dim> > melt_handler;
```

Comment:
I meant to physically move it to that place, so that these objects are kept together. You could still put a `private:` in front and `public:` after it :-)